### PR TITLE
Changed Chart to take a color array to describe a gradient

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed Chart to take a color array to describe a gradient

#### Where should the reviewer start?

Chart/doc.js

#### What testing has been done on this PR?

Storybook Chart Gradient story

#### How should this be manually tested?

Storybook Chart Gradient story

#### Do the grommet docs need to be updated?

automatic

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/Chart/Chart.js [hunk 1]
```diff
@@ -308,15 +310,68 @@ const Chart = React.forwardRef(
...
+    const viewBox = viewBounds.join(' ');
+    const colorName =
+      !useGradient && typeof color === 'object' ? color.color : color;
     const opacity = color.opacity
       ? theme.global.opacity[color.opacity]
       : undefined;
 
+    let stroke;
+    if (type !== 'point') {
+      if (useGradient) stroke = '#fff';
```
[the remark this answers is not in the document]
Comment (ericsoderberghp) [reply src/js/components/Chart/Chart.js]: good idea. done.

**Comment by ericsoderberghp** [comment]
I believe the console errors have been addressed.

File: src/js/components/Chart/Chart.js [hunk 2]
```diff
@@ -308,15 +310,68 @@ const Chart = React.forwardRef(
...
+      : [
+          -(strokeWidth / 2),
+          -(strokeWidth / 2),
+          size[0] + strokeWidth,
+          size[1] + strokeWidth,
+        ];
+    const viewBox = viewBounds.join(' ');
+    const colorName =
+      !useGradient && typeof color === 'object' ? color.color : color;
     const opacity = color.opacity
```
[the remark this answers is not in the document]
Comment (ericsoderberghp) [reply src/js/components/Chart/Chart.js]: Opacity has no effect when there is a gradient. We could potentially add it later when the need arises.

File: src/js/components/Chart/index.d.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ export interface ChartProps {
   gridArea?: GridAreaType;
   margin?: MarginType;
   bounds?: number[][];
-  color?: string | {color?: string,opacity?: "weak" | "medium" | "strong" | boolean};
+  color?: string | {color?: string,opacity?: "weak" | "medium" | "strong" | boolean} | {color?: string,value: number | number[]}[];
```
Comment (ShimiSun) [inline src/js/components/Chart/index.d.ts]:
```suggestion
  color?: string | {color?: string,opacity?: "weak" | "medium" | "strong" | boolean} | {color?: string, value?: number | number[]}[];
```

File: src/js/components/Chart/doc.js [hunk 4]
```diff
@@ -26,8 +26,17 @@ export const doc = Chart => {
...
+          color: PropTypes.string,
+          value: PropTypes.number,
+        }),
+      ),
     ])
-      .description('A color identifier to use for the graphic color.')
+      .description(
+        `A color identifier to use for the graphic color. If an
+      array is specified, it is used to create a gradient mask.`,
+      )
```
Comment (ShimiSun) [inline src/js/components/Chart/doc.js]: I would expect the description to explain what 'value' of color prop means.

**Comment by ShimiSun** [comment]
Screen Shot 2019-12-02 at 2 34 45 PM
I still see console errors.

File: src/js/components/Chart/index.d.ts (hunk 3 above)
Comment (ericsoderberghp) [reply src/js/components/Chart/index.d.ts]: Actually, both color and value are required if passing an array of objects. I've updated the typings to reflect this.

**Comment by ericsoderberghp** [comment]
Fixed console error.

**Pull request merged**